src/Unity.Mathematics/Extras/Plane.cs

```diff
@@ -0,0 +1,232 @@
+        /// <param name="coefficientB">Coefficient B from plane equation.</param>
+        /// <param name="coefficientC">Coefficient C from plane equation.</param>
+        /// <param name="coefficientD">Coefficient D from plane equation.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public Plane(float coefficientA, float coefficientB, float coefficientC, float coefficientD)
+        {
+            NormalAndDistance = Normalize(new float4(coefficientA, coefficientB, coefficientC, coefficientD));
+        }
+
+        /// <summary>
+        /// Constructs a plane with a normal vector and distance from the origin.
+        /// </summary>
+        /// <remarks>
+        /// The constructed plane will be the normalized form of the plane specified by the inputs.
+        /// </remarks>
+        /// <param name="normal">A non-zero vector that is perpendicular to the plane.  It may be any length.</param>
+        /// <param name="distance">Distance from the origin along the normal.  A negative value moves the plane in the
+        /// same direction as the normal while a positive value moves it in the opposite direction.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public Plane(float3 normal, float distance)
+        {
+            NormalAndDistance = Normalize(new float4(normal, distance));
+        }
+
+        /// <summary>
+        /// Constructs a plane with a normal vector and a point that lies in the plane.
+        /// </summary>
+        /// <remarks>
+        /// The constructed plane will be the normalized form of the plane specified by the inputs.
+        /// </remarks>
+        /// <param name="normal">A non-zero vector that is perpendicular to the plane.  It may be any length.</param>
+        /// <param name="pointInPlane">A point that lies in the plane.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public Plane(float3 normal, float3 pointInPlane)
+        : this(normal, -math.dot(normal, pointInPlane))
+        {
+        }
+
+        /// <summary>
+        /// Constructs a plane with two vectors and a point that all lie in the plane.
+        /// </summary>
+        /// <remarks>
+        /// The constructed plane will be the normalized form of the plane specified by the inputs.
+        /// </remarks>
+        /// <param name="vector1InPlane">A non-zero vector that lies in the plane.  It may be any length.</param>
+        /// <param name="vector2InPlane">A non-zero vector that lies in the plane.  It may be any length and must not be a scalar multiple of <paramref name="vector1InPlane"/>.</param>
+        /// <param name="pointInPlane">A point that lies in the plane.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public Plane(float3 vector1InPlane, float3 vector2InPlane, float3 pointInPlane)
+        : this(math.normalize(math.cross(vector1InPlane, vector2InPlane)), pointInPlane)
```

Comment:
Leaving a note for myself since this looks like it will hit a double normalization...